### PR TITLE
Update zh-TW.json

### PR DESCRIPTION
**Pull request by xatier**
- Update zh-TW translation.
- Sync with [locales/en.json](https://github.com/BoostIO/Boostnote/blob/e44381f295a8c6c7d2391bb7edab5cc9730c72a0/locales/en.json).

## Type of changes

- :white_circle: Bug fix (Change that fixed an issue)
- :white_circle: Breaking change (Change that can cause existing functionality to change)
- :white_circle: Improvement (Change that improves the code. Maybe performance or development improvement)
- :white_circle: Feature (Change that adds new functionality)
- :black_circle: Documentation change (Change that modifies documentation. Maybe typo fixes)

## Checklist:

- :white_circle: My code follows [the project code style](docs/code_style.md)
- :white_circle: I have written test for my code and it has been tested
- :white_circle: All existing tests have been passed
- :white_circle: I have attached a screenshot/video to visualize my change if possible
- :black_circle: This PR will modify the UI or affects the UX
- :white_circle: This PR will add/update/delete a keybinding

### CODE REVIEW

**Comment by xatier** [comment]
@opw0011 @PeterDaveHello would you like to help review my changes?

File: locales/zh-TW.json [hunk 1]
```diff
@@ -22,85 +23,93 @@
...
   "English": "English",
-  "German": "German",
-  "French": "French",
+  "German": "德文",
+  "French": "法文",
   "Show \"Saved to Clipboard\" notification when copying": "複製的時候，顯示 \"已複製到剪貼簿\" 的通知",
   "All Notes": "所有筆記",
-  "Starred": "星號收藏",
-  "Are you sure to ": "你確定要 ",
+  "Starred": "打星收藏",
```
Comment (PeterDaveHello) [inline locales/zh-TW.json:111]: This one seems strange

File: locales/zh-TW.json [hunk 2]
```diff
@@ -113,61 +122,76 @@
...
+  "Delete Note": "刪除模式",
   "Restore": "還原",
   "Permanent Delete": "永久刪除",
   "Confirm note deletion": "確認刪除筆記",
-  "This will permanently remove this note.": "這將會永久地刪除這條筆記",
-  "Successfully applied!": "設定成功",
-  "Albanian": "Albanian",
-  "Chinese (zh-CN)": "简体中文",
-  "Chinese (zh-TW)": "繁體中文",
+  "This will permanently remove this note.": "永久地刪除此筆記",
```
Comment (PeterDaveHello) [inline locales/zh-TW.json:145]: Previous one seems okay to me

File: locales/zh-TW.json [hunk 3]
```diff
@@ -113,61 +122,76 @@
...
-  "Enable HTML label in mermaid flowcharts": "Enable HTML label in mermaid flowcharts ⚠ This option potentially has a risk of XSS.",
-  "Wrap line in Snippet Note": "Wrap line in Snippet Note"
+  "Snippet Default Language": "程式碼片段預設語言",
+  "New notes are tagged with the filtering tags": "以過慮標籤標記新筆記",
+  "Show menu bar": "顯示功能列",
+  "Auto Detect": "自動偵測",
+  "Filter tags/folders...": "過濾標籤/資料夾...",
+  "Enable HTML label in mermaid flowcharts": "在 mermaid 流程圖中啟用 HTML 標籤 ⚠ 本選項有潛在的 XSS 安全風險。",
+  "Wrap line in Snippet Note": "Snippet Note 行尾換行",
+  "Enable Auto Update": "Enable Auto Update"
```
Comment (PeterDaveHello) [inline locales/zh-TW.json:196]: Missed translation?

File: locales/zh-TW.json [hunk 4]
```diff
@@ -113,61 +122,76 @@
...
   "Convert textual arrows to beautiful signs. ⚠ This will interfere with using HTML comments in your Markdown.": "將文本箭頭轉換為完整符號。 ⚠ 注意這會影響 Markdown 的 HTML 注釋。",
-  "Default New Note": "預設新筆記類型",
-  "Show only related tags": "只顯示相關標籤",
-  "Snippet Default Language": "程式碼片段預設語言",
-  "Disable Direct Write (It will be applied after restarting)": "停用直接編輯 (重啟後生效)",
+  "⚠ You have pasted a link referring an attachment that could not be found in the storage location of this note. Pasting links referring attachments is only supported if the source and destination location is the same storage. Please Drag&Drop the attachment instead! ⚠": "⚠ 您貼上了一個不存在本筆記儲存空間的附加檔案連結。貼上附加檔案連結功能只支援剪下貼上於相同儲存空間之間。請改以拖拉 Drag&Drop 附加檔案！⚠",
+  "Spellcheck disabled": "拼寫檢查已關閉",
+  "Save tags of a note in alphabetical order": "依照字母排序儲存標籤",
+  "Enable live count of notes": "啟用即時統計筆記數量",
   "Enable smart table editor": "啟用智能表格編輯器",
```
Comment (PeterDaveHello) [inline locales/zh-TW.json:188]: Due to the changes of context, I believe that this one should also be revised.

File: locales/zh-TW.json [hunk 5]
```diff
@@ -113,61 +122,76 @@
...
-  "Unsaved Changes!": "你必須儲存一下！",
-  "Russian": "Russian",
+  "Danish": "丹麥文",
+  "Japanese": "日文",
+  "Korean": "韓文",
+  "Norwegian": "挪威語",
+  "Polish": "波蘭文",
+  "Portuguese": "葡萄牙文",
+  "Spanish": "西班牙文",
+  "Unsaved Changes!": "您必須儲存一下！",
```
Comment (PeterDaveHello) [inline locales/zh-TW.json:158]: This one may also needs to be revised

File: locales/zh-TW.json [hunk 6]
```diff
@@ -22,85 +23,93 @@
...
   "Add Folder": "新增資料夾",
+  "Select Folder": "選擇資料夾",
   "Open Storage folder": "開啟儲存資料夾",
   "Unlink": "解除連結",
   "Edit": "編輯",
   "Delete": "刪除",
-  "Interface": "界面",
-  "Interface Theme": "主題",
+  "Interface": "介面",
+  "Interface Theme": "介面主題",
```
Comment (PeterDaveHello) [inline locales/zh-TW.json:37]: The previous one looks fine to me.

File: locales/zh-TW.json (hunk 1 above)
Comment (xatier) [reply locales/zh-TW.json:111]: what about "我的最愛"? A bit old fashioned though lol

File: locales/zh-TW.json (hunk 2 above)
Comment (xatier) [reply locales/zh-TW.json:145]: It sounds a bit too orally imo.

File: locales/zh-TW.json (hunk 3 above)
Comment (xatier) [reply locales/zh-TW.json]: Nice catch. Fixed.

File: locales/zh-TW.json (hunk 6 above)
Comment (xatier) [reply locales/zh-TW.json:37]: 介面 is more Taiwan-ish.

https://www.moedict.tw/~%E4%BB%8B%E9%9D%A2

File: locales/zh-TW.json (hunk 5 above)
Comment (xatier) [reply locales/zh-TW.json:158]: Updated.

File: locales/zh-TW.json (hunk 4 above)
Comment (xatier) [reply locales/zh-TW.json:188]: Any suggestions?

File: locales/zh-TW.json (hunk 2 above)
Comment (xatier) [reply locales/zh-TW.json:145]: Maybe "永久刪除此筆記" sound better?

File: locales/zh-TW.json [hunk 7]
```diff
@@ -22,85 +23,93 @@
...
   "Editor Keymap": "編輯器 Keymap",
   "default": "預設",
   "vim": "vim",
   "emacs": "emacs",
-  "⚠️ Please restart boostnote after you change the keymap": "⚠️ 修改鍵盤配置請重新開啟 Boostnote ",
+  "⚠️ Please restart boostnote after you change the keymap": "⚠️ 修改鍵盤配置後請重新啟動 Boostnote ",
   "Show line numbers in the editor": "在編輯器中顯示行號",
   "Allow editor to scroll past the last line": "允許編輯器捲軸捲動超過最後一行",
-  "Bring in web page title when pasting URL on editor": "在編輯器貼上網址的時候，自動加上網頁標題",
+  "Enable smart quotes": "啟用智慧引號",
```
Comment (Brli) [inline locales/zh-TW.json:62]: What does it really do? Automatically add right quotes when input left quotes?
Maybe "啟用下引號補完" to describe it?

File: locales/zh-TW.json [hunk 8]
```diff
@@ -113,61 +122,76 @@
...
-  "⚠ You have pasted a link referring an attachment that could not be found in the storage location of this note. Pasting links referring attachments is only supported if the source and destination location is the same storage. Please Drag&Drop the attachment instead! ⚠": "⚠ You have pasted a link referring an attachment that could not be found in the storage location of this note. Pasting links referring attachments is only supported if the source and destination location is the same storage. Please Drag&Drop the attachment instead! ⚠",
-  "Spellcheck disabled": "Spellcheck disabled",
-  "Show menu bar": "Show menu bar",
-  "Auto Detect": "Auto Detect",
-  "Filter tags/folders...": "filter tags/folders...",
-  "Enable HTML label in mermaid flowcharts": "Enable HTML label in mermaid flowcharts ⚠ This option potentially has a risk of XSS.",
-  "Wrap line in Snippet Note": "Wrap line in Snippet Note"
+  "Snippet Default Language": "程式碼片段預設語言",
+  "New notes are tagged with the filtering tags": "以過慮標籤標記新筆記",
+  "Show menu bar": "顯示功能列",
```
Comment (Brli) [inline locales/zh-TW.json:191]: "顯示選單列"？
功能列通常是下方顯示功能說明的地方

File: locales/zh-TW.json [hunk 9]
```diff
@@ -113,61 +122,76 @@
...
-  "Render newlines in Markdown paragraphs as <br>": "在 Markdown 段落中使用 <br> 換行",
-  "⚠ You have pasted a link referring an attachment that could not be found in the storage location of this note. Pasting links referring attachments is only supported if the source and destination location is the same storage. Please Drag&Drop the attachment instead! ⚠": "⚠ You have pasted a link referring an attachment that could not be found in the storage location of this note. Pasting links referring attachments is only supported if the source and destination location is the same storage. Please Drag&Drop the attachment instead! ⚠",
-  "Spellcheck disabled": "Spellcheck disabled",
-  "Show menu bar": "Show menu bar",
-  "Auto Detect": "Auto Detect",
-  "Filter tags/folders...": "filter tags/folders...",
-  "Enable HTML label in mermaid flowcharts": "Enable HTML label in mermaid flowcharts ⚠ This option potentially has a risk of XSS.",
-  "Wrap line in Snippet Note": "Wrap line in Snippet Note"
+  "Snippet Default Language": "程式碼片段預設語言",
+  "New notes are tagged with the filtering tags": "以過慮標籤標記新筆記",
```
Comment (Brli) [inline locales/zh-TW.json:190]: 濾

File: locales/zh-TW.json [hunk 10]
```diff
@@ -113,61 +122,76 @@
...
+  "Show menu bar": "顯示功能列",
+  "Auto Detect": "自動偵測",
+  "Filter tags/folders...": "過濾標籤/資料夾...",
+  "Enable HTML label in mermaid flowcharts": "在 mermaid 流程圖中啟用 HTML 標籤 ⚠ 本選項有潛在的 XSS 安全風險。",
```
Comment (Brli) [inline locales/zh-TW.json:194]: 警告段落不存在於原文？
是新發現嗎還是有ticket?

File: locales/zh-TW.json [hunk 11]
```diff
@@ -22,85 +23,93 @@
...
+  "You can choose to enable or disable this option.": "您可以選擇啟用或停用這項功能",
+  "Enable analytics to help improve Boostnote": "啟用數據分析以協助我們改進 Boostnote",
   "Crowdfunding": "群眾募資",
-  "Dear Boostnote users,": "親愛的用戶：",
-  "Thank you for using Boostnote!": "謝謝你使用 Boostnote！",
+  "Dear Boostnote users,": "親愛的使用者：",
+  "Thank you for using Boostnote!": "感謝您使用 Boostnote！",
   "Boostnote is used in about 200 different countries and regions by an awesome community of developers.": "大約有 200 個不同的國家和地區的優秀開發者們都在使用 Boostnote！",
-  "To support our growing userbase, and satisfy community expectations,": "為了繼續支持這種發展，和滿足社群的期待，",
+  "To support our growing userbase, and satisfy community expectations,": "為了繼續支持我們的使用者成長與滿足社群期待，",
```
Comment (Brli) [inline locales/zh-TW.json:98]: 為了維持使用者基數的增長以及滿足社群的期待

File: locales/zh-TW.json [hunk 12]
```diff
@@ -22,85 +23,93 @@
...
-  "Bring in web page title when pasting URL on editor": "在編輯器貼上網址的時候，自動加上網頁標題",
+  "Enable smart quotes": "啟用智慧引號",
+  "Bring in web page title when pasting URL on editor": "在編輯器貼上網址時自動加上網頁標題",
   "Preview": "預覽頁面",
   "Preview Font Size": "預覽頁面字型大小",
   "Preview Font Family": "預覽頁面字體",
   "Code Block Theme": "程式碼區塊主題",
   "Allow preview to scroll past the last line": "允許預覽頁面捲軸捲動超過最後一行",
-  "Show line numbers for preview code blocks": "在預覽頁面的程式碼區塊中顯示行號",
+  "Show line numbers for preview code blocks": "在程式碼區塊預覽中顯示行號",
```
Comment (Brli) [inline locales/zh-TW.json:69]: 不太確定預覽頁面是不是也有分區塊
是的話說不定原本的比較適合

File: locales/zh-TW.json [hunk 13]
```diff
@@ -113,61 +122,76 @@
...
   "Only allow secure html tags (recommended)": "只允許安全的 HTML 標籤 (建議)",
+  "Render newlines in Markdown paragraphs as <br>": "在 Markdown 段落中使用 <br> 換行",
   "Allow styles": "允許樣式",
   "Allow dangerous html tags": "允許危險的 HTML 標籤",
   "Convert textual arrows to beautiful signs. ⚠ This will interfere with using HTML comments in your Markdown.": "將文本箭頭轉換為完整符號。 ⚠ 注意這會影響 Markdown 的 HTML 注釋。",
-  "Default New Note": "預設新筆記類型",
-  "Show only related tags": "只顯示相關標籤",
-  "Snippet Default Language": "程式碼片段預設語言",
-  "Disable Direct Write (It will be applied after restarting)": "停用直接編輯 (重啟後生效)",
+  "⚠ You have pasted a link referring an attachment that could not be found in the storage location of this note. Pasting links referring attachments is only supported if the source and destination location is the same storage. Please Drag&Drop the attachment instead! ⚠": "⚠ 您貼上了一個不存在本筆記儲存空間的附加檔案連結。剪下貼上附加檔案連結功能只支援於相同儲存空間之間。請改以拖拉 Drag&Drop 附加檔案！⚠",
```
Comment (Brli) [inline locales/zh-TW.json:184]: …只支援相同儲存空間之「內」…

File: locales/zh-TW.json [hunk 14]
```diff
@@ -113,61 +122,76 @@
...
+  "Spellcheck disabled": "拼寫檢查已關閉",
+  "Save tags of a note in alphabetical order": "依照字母排序儲存標籤",
+  "Enable live count of notes": "啟用即時統計筆記數量",
```
Comment (Brli) [inline locales/zh-TW.json:187]: 這個功能是算字還是算筆記數？沒操作軟體不是很清楚

File: locales/zh-TW.json [hunk 15]
```diff
@@ -113,61 +122,76 @@
...
-  "Filter tags/folders...": "filter tags/folders...",
-  "Enable HTML label in mermaid flowcharts": "Enable HTML label in mermaid flowcharts ⚠ This option potentially has a risk of XSS.",
-  "Wrap line in Snippet Note": "Wrap line in Snippet Note"
+  "Snippet Default Language": "程式碼片段預設語言",
+  "New notes are tagged with the filtering tags": "以過慮標籤標記新筆記",
+  "Show menu bar": "顯示功能列",
+  "Auto Detect": "自動偵測",
+  "Filter tags/folders...": "過濾標籤/資料夾...",
+  "Enable HTML label in mermaid flowcharts": "在 mermaid 流程圖中啟用 HTML 標籤 ⚠ 本選項有潛在的 XSS 安全風險。",
+  "Wrap line in Snippet Note": "Snippet Note 行尾換行",
```
Comment (Brli) [inline locales/zh-TW.json:195]: 這是自動換行功能嗎？
或許可以翻成「在 Snippest Note 中啟用自動換行」？

File: locales/zh-TW.json (hunk 7 above)
Comment (xatier) [reply locales/zh-TW.json:62]: not quite, I believe it's for the typographer feature in `markdownit` [1]. see demo [2]

[1] https://github.com/BoostIO/Boostnote/blob/592aca1539166bb9a6d1eaf36a20203fff584c37/browser/lib/markdown.js#L29 
[2] https://markdown-it.github.io/

File: locales/zh-TW.json (hunk 8 above)
Comment (xatier) [reply locales/zh-TW.json]: Fixed.

File: locales/zh-TW.json (hunk 9 above)
Comment (xatier) [reply locales/zh-TW.json:190]: Fixed.

File: locales/zh-TW.json (hunk 11 above)
Comment (xatier) [reply locales/zh-TW.json:98]: Fixed.

File: locales/zh-TW.json (hunk 12 above)
Comment (xatier) [reply locales/zh-TW.json:69]: reverted.

File: locales/zh-TW.json (hunk 13 above)
Comment (xatier) [reply locales/zh-TW.json:184]: Fixed.

File: locales/zh-TW.json (hunk 14 above)
Comment (xatier) [reply locales/zh-TW.json:187]: live count of notes when selecting tags
https://github.com/BoostIO/Boostnote/pull/2314

File: locales/zh-TW.json (hunk 15 above)
Comment (xatier) [reply locales/zh-TW.json:195]: Fixed.
I was wondering to pick between 換行 vs 折行, how do you think?

File: locales/zh-TW.json (hunk 4 above)
Comment (Brli) [reply locales/zh-TW.json:188]: 啟用智慧型表格編輯器

File: locales/zh-TW.json (hunk 14 above)
Comment (Brli) [reply locales/zh-TW.json:187]: got it

File: locales/zh-TW.json (hunk 15 above)
Comment (Brli) [reply locales/zh-TW.json]: It's just personal reference, I'd respect your choice

File: locales/zh-TW.json (hunk 7 above)
Comment (Brli) [reply locales/zh-TW.json:62]: So, it is like replace " to “ or ” in the line?
Then, can be like "啟用自動引號轉換"?

Comment (Brli) [reply locales/zh-TW.json:62]: What about "啟用引號自動變換"?

Comment (xatier) [reply locales/zh-TW.json:62]: sure.

**Comment by xatier** [comment]
I've fixed most of the suggestions above.

File: locales/zh-TW.json (hunk 4 above)
Comment (PeterDaveHello) [reply locales/zh-TW.json:188]: At least the term of smart should be revised?

File: locales/zh-TW.json (hunk 6 above)
Comment (PeterDaveHello) [reply locales/zh-TW.json:37]: I mean the prefix of theme may not need to be attached?

File: locales/zh-TW.json (hunk 1 above)
Comment (PeterDaveHello) [reply locales/zh-TW.json]: Sounds better

**Comment by PeterDaveHello** [comment]
Maybe also squash the commits after final fixes?

File: locales/zh-TW.json (hunk 4 above)
Comment (xatier) [reply locales/zh-TW.json:188]: I don't have strong opinions here, from the code it appears to enable the editor activity handler. I'm not exactly sure how "smart" it is, but would like to stay with the original if we don't have other better alternatives.

https://github.com/BoostIO/Boostnote/blob/0d797ce8a86c2701bda332c97af34e7fd0d1bb07/browser/main/modals/PreferencesModal/UiTab.js#L817
https://github.com/BoostIO/Boostnote/blob/592aca1539166bb9a6d1eaf36a20203fff584c37/browser/components/CodeEditor.js#L491

Comment (PeterDaveHello) [reply locales/zh-TW.json:188]: I think "智能" should be "智慧"

Comment (xatier) [reply locales/zh-TW.json:188]: Uh, you were a bit too late, this commit has been merged. You may submit another PR for such change.

**Pull request merged**